app/src/main/java/com/andrew/tasky/agenda/data/agenda/AgendaRepositoryImpl.kt

```diff
@@ -212,54 +212,75 @@ class AgendaRepositoryImpl(
         taskRepository.uploadCreateAndUpdateModifiedTasks()
         eventRepository.uploadCreateAndUpdateModifiedEvents()
 
-        val reminderDeleteIds = db.getReminderDao().getModifiedReminders().first().filter {
-            it.modifiedType == ModifiedType.DELETE
-        }.map {
-            it.id
-        }
-        val taskDeleteIds = db.getTaskDao().getModifiedTasks().first().filter {
-            it.modifiedType == ModifiedType.DELETE
-        }.map {
-            it.id
-        }
-        val eventDeleteIds = db.getEventDao().getModifiedEvents().first().filter {
-            it.modifiedType == ModifiedType.DELETE
-        }.map {
-            it.id
-        }
-
-        if (reminderDeleteIds.isNotEmpty() ||
-            taskDeleteIds.isNotEmpty() ||
-            eventDeleteIds.isNotEmpty()
-        ) {
-            val syncAgendaRequest = SyncAgendaRequest(
-                eventDeleteIds,
-                taskDeleteIds,
-                reminderDeleteIds
-            )
-            val results = getResourceResult { agendaApi.syncAgendaItems(syncAgendaRequest) }
-            when (results) {
-                is Resource.Error -> {
-                    Log.e(
-                        "SyncAgendaItem Error",
-                        results.message?.asString(appContext) ?: "Unknown Error"
-                    )
+        val coroutine = CoroutineScope(Dispatchers.IO + SupervisorJob())
+        return coroutine.async {
```

Comment:
I'd avoid using custom scopes in components that don't have a lifecycle (such as this repo - it's a singleton). Instead I'd rely on using `viewModelScope` and making this a suspend function. You can then still use async this way:
```
supervisorScope {
    async { ... }
}

```